lib/update/updater.js

```diff
@@ -19,7 +20,11 @@ var Updater = function (config) {
   this.extractor = require('./extractor').factory(this.config.extractor)
   this.downloading = false
   this.ack = null
+
   this.deviceId = this._fetchDeviceId()
+  if (this.deviceId == null) {
```

Comment:
``if (!this.deviceId)`` is probably safer, or consider throw and catching an exception.